lib/rubocop/cop/style/redundant_fetch_block.rb

```diff
@@ -82,6 +81,16 @@ def const_type?(node)
           node&.const_type?
         end
 
+        def should_not_check?(send, body)
+          (body&.const_type? && !check_for_constant?) ||
+            (body&.str_type? && !check_for_string?) ||
+            rails_cache?(send.receiver)
```

Comment:
I was late for the discussion of #8196.

I agree with the following @bbatsov's opinion.

> Fixing this is going to be tricky, due to the dynamic nature of Ruby - we can never be sure about the type of a receiver. The best thing that comes to mind is to have some configuration about allowed receivers or whatever, as I don't want to hardcode some special handling of Rails.cache. //cc @fatkodima

Ideally RuboCop core should be Rails unaware because RuboCop core focuses on Ruby. It is better to create `IgnoredReceivers` option and configure it with RuboCop Rails (or users).

Reply:
I think we all agree as to what is the best. I see this PR as a temporary improvement, before a `IgnoredReceivers` option.

Reply:
Yeah, I get it! I'll merge this PR.